### PR TITLE
Compact: Replace group with resolution in compact metrics.

### PR DESCRIPTION
**Pull request by SuperQ**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Compaction metrics have too high a cardinality, causing metric bloat on
large installations. The group information is better suited to logs.
* Replace with a `resolution` label to the compaction counters.

Fixes: https://github.com/thanos-io/thanos/issues/5841

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
Changes make sense to me. Thanks. We just need to fix the test

**Comment by SuperQ** [comment]
Yea, I worked on the tests a bit, but ran out of time this week. Will see if I can figure out the rest soon.

**Comment by SuperQ** [comment]
Ok, I updated this to fix the tests and also update the compact dashboards.

**Comment by yeya24** [comment]
Is the CircleCI failure related to this pr? It would be great if you could take a look

**Comment by SuperQ** [comment]
Yes, I realized I need to refactor all the tests to use the new resolution label, not group. Will work on it soon.

**Comment by SuperQ** [comment]
Ok, much closer on the updates to the tests.

**Comment by SuperQ** [comment]
Ping @yeya24, tests are passing now.

**Review by yeya24 — DISMISSED** [review]
Thanks! I will let other maintainers take another look before merging as it is a breaking change.

**Comment by SuperQ** [comment]
Yes, this would be great to get in before the next release. I have one cluster with 90,000 metrics on the compact that is nearly useless to query. Plus others with 10k+.

**Comment by SuperQ** [comment]
Maybe @metalmatze or @GiedriusS ?

File: CHANGELOG.md
Comment (bwplotka) [inline CHANGELOG.md:39]:
```suggestion
- [#6049](https://github.com/thanos-io/thanos/pull/6049) Compact: *breaking :warning:* Replace group with resolution in compact metrics to avoid cardinality explosion on compact metrics for large numbers of groups.
```

**Review by bwplotka — DISMISSED** [review]
Hm, so generally the groups are useful, as they allow checking each group progress. However indeed for large numbers of streams (e.g. 1k+) this gets troubling. I think it's fair to fix the cardinality explosion problem ASAP and merge this, just I see users with smaller number of groups who liked previous approach. I think we can follow up with opt-in grouping later on if needed 👍🏽

File: CHANGELOG.md
Comment (SuperQ) [reply CHANGELOG.md]: Updated.

**Review by yeya24 — APPROVED** [review]

**Comment by SuperQ** [comment]
Can this be merged?

**Pull request merged**
